File: 30_days_of_code/day10/day10.py
#!/bin/python3


def maxConsecutiveOnes(n):
    count = 0
    while n != 0:
        n = (n & (n << 1))
        count += 1

    return count


if __name__ == '__main__':
    n = int(input())
    print(maxConsecutiveOnes(n))

File: 30_days_of_code/day10/day10.c
#include <stdio.h>
#include <stdlib.h>

int maxConsecutiveOnes(int n)
{
  int count = 0;
  while (n != 0) {
    n = (n & (n << 1));
    ++count;
  }

  return count;
}

int main()
{
  int n;
  scanf("%d", &n);

  printf("%d\n", maxConsecutiveOnes(n));
  return 0;
}
